From 257501226bbc9c68ca785eb44d06fba2776ec742 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@amyspark.me>
Date: Thu, 17 Nov 2022 11:56:38 -0300
Subject: [PATCH 14/18] meson: add (commented out) support for MASM

Reason for dummying it out is because it is orders of magnitude slower than the equivalent C++ build.
---
 source/data/meson.build | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/source/data/meson.build b/source/data/meson.build
index ce6f3cb..342952d 100644
--- a/source/data/meson.build
+++ b/source/data/meson.build
@@ -10,7 +10,14 @@ endif
 
 if host_machine.system() == 'windows'
   if cpp.get_argument_syntax() == 'msvc'
-    icudata_asm_output = '@0@.c'.format(U_ICUDATA_NAME)
+    ## Uncomment if you want to use MASM (>= 0.64), but be aware
+    ## it is Very Slow to build.
+    # if add_languages('masm', required: false)
+    #   icudata_asm_output = '@0@.masm'.format(U_ICUDATA_NAME)
+    #   icudata_command += ['-a', 'masm']
+    # else
+      icudata_asm_output = '@0@.c'.format(U_ICUDATA_NAME)
+    # endif
   else
     icudata_asm_output = '@0@.S'.format(U_ICUDATA_NAME)
     icudata_command += ['-a', 'gcc-mingw64']
-- 
2.37.1.windows.1

